src/bflib_sprfnt.c

```diff
@@ -1920,13 +1920,14 @@ short dbc_initialize(const char *fpath)
     char fname[DISKPATH_SIZE];
     if ((fpath != NULL) && (fpath[0] != 0))
     {
-      strncpy(fname, fpath, DISKPATH_SIZE);
-      fname[DISKPATH_SIZE-1] = '\0';
-      if (fname[strlen(fname)-1] != '/')
-        strcat(fname, "/");
+        snprintf(fname, DISKPATH_SIZE, "%s", fpath);
+        if (fname[strlen(fname) - 1] != '/')
+        {
+            strcat(fname, "/");
```

Comment:
There are errors here:
Imagine `DISKPATH_SIZE = 4` and `fpath='path'`
fname before strcat is `pat0` (0 is end of string, `h` is truncated)

strlen goes 3 and `fname[2]` is `t`.

After strcat will be overflow 'pat/0' where 0 is out of bounds

Reply:
There is also the `strcat(fname, active_dbcfont->fname);` later on. Its basically performing `"fname/active_dbcfont->fname"` when fname is not empty in a very convoluted way. In the past path names were length-limited but this is not the case anymore and you have to allocate a buffer on the heap.